src/antsibull/collection_meta.py

```diff
@@ -0,0 +1,54 @@
+# Author: Felix Fontein <felix@fontein.de>
+# Author: Toshio Kuratomi <tkuratom@redhat.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Ansible Project, 2020
+
+"""
+Classes to encapsulate collection metadata from collection-meta.yaml
+"""
+import typing as t
+import os
+
+from antsibull_core.yaml import load_yaml_file
+
+
+class CollectionMetadata:
+    '''
+    Stores metadata about one collection.
+    '''
+
+    changelog_url: t.Optional[str]
+
+    def __init__(self, source: t.Optional[t.Mapping[str, t.Any]] = None):
+        if source is None:
+            source = {}
+        self.changelog_url: t.Optional[str] = source.get('changelog-url')
+        self.collection_directory: t.Optional[str] = source.get('collection-directory')
+        self.repository: t.Optional[str] = source.get('repository')
```

Comment:
I would remove the types here, since they are already shown above:
```suggestion
        self.changelog_url = source.get('changelog-url')
        self.collection_directory = source.get('collection-directory')
        self.repository = source.get('repository')
```